src/generators.rs

```diff
@@ -950,6 +952,94 @@ pub fn binomial_tree_graph(
     })
 }
 
+///--------------------
+
+/// Creates a full r-ary tree of `n` nodes.
+/// Sometimes called a k-ary, n-ary, or m-ary tree.
+///
+/// :param int order: Order of the tree.
+/// :param list weights: A list of node weights. If the number of weights is
+///     less than 2**order extra nodes with with None will be appended.
```

Comment:
I do not think it is supposed to be 2 here, shouldn't it be related to `r` instead?